src/org/zaproxy/zap/extension/websocket/ui/httppanel/models/StringWebSocketPanelViewModel.java

```diff
@@ -36,7 +36,7 @@ public String getData() {
             data = "";
         } else {
 	        try {
-				data = webSocketMessage.getReadablePayload();
+				data = webSocketMessage.getPayloadAsString();

```

Comment:
This breaks the current handling of binary payloads (`InvalidUtf8Exception` is no longer thrown so we end up replacing its contents, with potentially malformed data). When the view is editable we should use the older method.

Reply:
The only one case which is editable, it is in the Websocket Manual Editor, right?

Reply:
It can also be edited in the Break panel.